service/src/main/java/org/jboss/sbomer/service/feature/sbom/errata/event/AdvisoryEventUtils.java

```diff
@@ -139,19 +143,22 @@ public static Set<String> createPurls(
             List<RepositoryCoordinates> repositories,
             String version,
             boolean includeRepositoryQualifiers) {
-
-        return repositories.stream()
+        Set<String> set = repositories.stream()
                 .map(repository -> createPurl(repository, version, includeRepositoryQualifiers))
                 .filter(Objects::nonNull)
-                .sorted(Comparator.comparingInt(String::length).reversed()) // longest first
-                .collect(Collectors.toSet());
+                .sorted(
+                        Comparator.comparingInt(String::length)
+                                .reversed() // longest first
+                                .thenComparing(Comparator.naturalOrder()))
+                .collect(Collectors.toCollection(LinkedHashSet::new));
+        return Collections.unmodifiableSet(set);
```

Comment:
I fixed this to use `LinkedHashSet` so that it's actually sorted.

This change also affects the test (that actually test the incorrect ordering).

I added the `.thenComparing(Comparator.naturalOrder()))` to handle strings of the same length so that we get a predictable ordering.

Reply:
Sounds good, thanks!